tests/lib/managers/folders-test.js

```diff
@@ -276,4 +276,19 @@ describe('Folders', function() {
 		});
 	});
 
+	describe('restoreFolder()', function() {
+
+		it('should make POST call to restore a folder', function() {
+			sandbox.stub(boxClientFake, 'defaultResponseHandler');
+			sandbox.mock(boxClientFake).expects('post').withArgs('/folders/1234');
+			folders.restoreFolder(FOLDER_ID, 'Folder Restored', 0);
+		});
+
+		it('should call BoxClient defaultResponseHandler method with the callback when response is returned', function(done) {
+			sandbox.mock(boxClientFake).expects('defaultResponseHandler').withArgs(done).returns(done);
+			sandbox.stub(boxClientFake, 'post').yieldsAsync();
+			folders.restoreFolder(FOLDER_ID, 'Folder Restored', 0, done);
+		});
+	});
```

Comment:
I did add values as part of the test. Do you want to add tests for different values for the name and / or parentFolderId?